src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -1531,8 +1546,8 @@ trait Contexts { self: Analyzer =>
           return ambiguousDefnAndImport(defSym.alternatives.head.owner, imp1)
       })
 
-      // If the defSym is at 4, and there is a def at 1 in scope, then the reference is ambiguous.
-      if (foreignDefined && !defSym.isPackage) {
+      // If the defSym is at 4, and there is a def at 1 in scope due to packaging, then the reference is ambiguous.
+      if (foreignDefined && !defSym.isPackage && !thisContext.unit.isJava) {
```

Comment:
Since Java does not have nested packagings for visibility, this extra check is unnecessary in Java.

The level one definition in Scala would have to be defined in an enclosing package in the same file. That is, the definition found at level 4 is in a more nested package, but is from a different file.